lib/statsd/instrument/backends/store_backend.rb

```diff
@@ -0,0 +1,73 @@
+require 'monitor'
+require 'time'
+require 'fileutils'
+
+module StatsD::Instrument::Backends
+  class StoreBackend < StatsD::Instrument::Backend
+    MAXIMUM_NAME_LENGTH = 256
+    TYPE_PACKS = {
+      c:  'Qq',
+      ms: 'Qf',
+      g:  'Qf',
+      h:  'Qf',
+    }
+    MetricNameStruct = Struct.new(:id, :name, :type) do
+      def to_bin
+        raise "Metric name cannot be longer than #{MAXIMUM_NAME_LENGTH}" if name.size > MAXIMUM_NAME_LENGTH
+        [id, name, type.to_s].pack("QA#{MAXIMUM_NAME_LENGTH}A2")
+      end
+    end
+    MetricStruct = Struct.new(:id, :value, :type) do
```

Comment:
Currently measurement is only 6 bytes and I want to keep on that, but I think I know how to store sample rate in separate record type (it does not change often right?) so it will only be stored when it's changed.